android/greengrass-nucleus-android/src/main/java/com/aws/greengrass/android/provision/BaseProvisionManager.java

```diff
@@ -0,0 +1,452 @@
+            // Access HashMaps and ArrayList by key(s)
+            HashMap system = (HashMap) yamlMap.get(DeviceConfiguration.SYSTEM_NAMESPACE_KEY);
+            thingName = (String) system.get(DeviceConfiguration.DEVICE_PARAM_THING_NAME);
+        } catch (FileNotFoundException e) {
+            logger.atWarn().log("Couldn't find {} file.", CONFIG_YAML_FILE);
+        } catch (NoSuchFileException e) {
+            logger.atWarn().log("File {} doesn't exist.", CONFIG_YAML_FILE);
+        } catch (Exception e) {
+            logger.atWarn().log("File {} doesn't have thing name.",
+                    CONFIG_YAML_FILE);
+        }
+        return thingName;
+    }
+
+    @NonNull
+    private ArrayList<String> generateArguments() {
+        ArrayList<String> argsList = new ArrayList<>();
+        Iterator<String> keys = config.fieldNames();
+
+        while (keys.hasNext()) {
+            String key = keys.next();
+            if (key.startsWith("-")) {
+                argsList.add(key);
+                argsList.add(config.get(key).asText());
+            }
+        }
+        argsList.add("--provision");
+        argsList.add("true");
+
+        return argsList;
+    }
+
+    /**
+     * Setup System properties.
+     *
+     * @throws Exception if the config is not valid
+     */
+    private void setupSystemProperties() throws Exception {
+        if (!config.has(PROVISION_ACCESS_KEY_ID)) {
+            logger.atError().log("Key {} is absent in the config file.", PROVISION_ACCESS_KEY_ID);
+            throw new Exception(String.format("Parameters do not contain \"%s\" key", PROVISION_ACCESS_KEY_ID));
+        } else {
+            System.setProperty(PROVISION_ACCESS_KEY_ID, config.get(PROVISION_ACCESS_KEY_ID).asText());
+        }
+        if (!config.has(PROVISION_SECRET_ACCESS_KEY)) {
+            logger.atError()
+                    .log(String.format("Key {} is absent in the config file.", PROVISION_SECRET_ACCESS_KEY));
+            throw new Exception(String.format("Parameters do not contain \"%s\" key", PROVISION_SECRET_ACCESS_KEY));
+        } else {
+            System.setProperty(PROVISION_SECRET_ACCESS_KEY, config.get(PROVISION_SECRET_ACCESS_KEY).asText());
+        }
+        if (config.has(PROVISION_SESSION_TOKEN)) {
+            System.setProperty(PROVISION_SESSION_TOKEN, config.get(PROVISION_SESSION_TOKEN).asText());
+        }
+    }
+
+    /**
+     * Remove all configuration files except system configuration.
+     *
+     * @param fileOrDirectory directory with configuration files
+     */
+    private void deleteConfigRecursive(@NonNull File fileOrDirectory) {
+        File[] list;
+        if (fileOrDirectory.isDirectory()) {
+            list = fileOrDirectory.listFiles();
+            if (list != null) {
+                for (File child : list) {
+                    if (child.getName().equals(CONFIG_YAML_FILE)) {
+                        continue;
+                    }
+                    child.delete();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.